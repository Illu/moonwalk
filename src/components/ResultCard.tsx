import React from "react";
import styled from "styled-components/native";

const Wrapper = styled.TouchableOpacity`
  padding: 20px;
  height: 100px;
  width: 100%;
  justify-content: center;
  border-bottom-width: 1px;
  background: ${({ theme }) => theme.colors.secondary};
  border-color: ${({ theme }) => theme.colors.uiAccent};
`;

const Title = styled.Text`
  font-weight: bold;
  color: ${({ theme }) => theme.colors.text};
  margin-bottom: 5px;
`;

const Subtitle = styled.Text`
  color: ${({ theme }) => theme.colors.text};
`;

const ResultCard = ({ data, showDetails }) => {
  const { name, net } = data;

  return (
    <Wrapper onPress={() => showDetails(data)}>
      <Title>{name}</Title>
      <Subtitle>{new Date(net).toLocaleString()}</Subtitle>
    </Wrapper>
  );
};

export default ResultCard;
