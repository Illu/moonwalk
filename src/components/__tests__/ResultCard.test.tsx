import React from "react";

import TestRenderer from "../../helpers/testRenderer";
import ResultCard from "../ResultCard";

const localeSpy = jest.spyOn(Date, "toLocaleString");
localeSpy.mockImplementation(jest.fn(() => "1/1/2020, 20:00:00"));

describe("ResultCard", () => {
  it("renders correctly", () => {
    const props = {
      data: {
        name: "$_TEST_NAME_$",
        net: "2020-08-05T02:00:00Z",
      },
      showDetails: jest.fn(),
    };
    const tree = TestRenderer(<ResultCard {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
