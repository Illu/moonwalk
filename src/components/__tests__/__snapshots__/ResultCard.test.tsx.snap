// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultCard renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "borderBottomWidth": 1,
      "borderColor": "#c3c4c6",
      "height": 100,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 20,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 20,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontWeight": "bold",
          "marginBottom": 5,
        },
      ]
    }
  >
    $_TEST_NAME_$
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
      ]
    }
  >
    8/4/2020, 10:00:00 PM
  </Text>
</View>
`;
